### PR TITLE
fix(tsconfig): empty tsconfig.json file does not cause error

### LINKED ISSUE #10733: Error while parsing empty tsconfig.json
Hi :)

While trying to run deno using: `deno run -c tsconfig.json main.js` I got:
```
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', cli/config_file.rs:290:59
```

The Error happens with a fully commented tsconfig.json file or an empty one.

#### How typescript and ts-node handle that ?
Running `ts-node -P tsconfig.json main.ts` or `tsc -p tsconfig.json main.ts` with an empty tsconfig.json works 

#### Potential fix
Return a valid empty json object in cli/config_file.rs:290 ?

```diff
-  let jsonc = jsonc_parser::parse_to_serde_value(text)?.unwrap();
+ let jsonc = jsonc_parser::parse_to_serde_value(text)?.unwrap_or(json!({}));
```

I'm ready to submit a PR if you want :)

*deno version*: 1.10.2

### PR DESCRIPTION
**Pull request by romainPrignon**
closes #10733

### CODE REVIEW

File: cli/config_file.rs [hunk 1]
```diff
@@ -293,7 +293,7 @@ impl ConfigFile {
   }
 
   pub fn new(text: &str, path: &Path) -> Result<Self, AnyError> {
-    let jsonc = jsonc_parser::parse_to_serde_value(text)?.unwrap();
+    let jsonc = jsonc_parser::parse_to_serde_value(text)?.unwrap_or(json!({}));
```
Comment (nayeemrmn) [inline cli/config_file.rs:296]: This should error just like the import map would:
```suggestion
    let jsonc = match jsonc_parser::parse_to_serde_value(text) {
      Ok(Some(value)) if value.is_object() => value,
      _ => return Err(anyhow!("Unable to pass config file JSON")),
    };
```

Comment (romainPrignon) [reply cli/config_file.rs:296]: ok, I didn't know how import map was handled. It makes more sense.

Comment (romainPrignon) [reply cli/config_file.rs:296]: I'll apply the fix and update the test accordingly

File: cli/config_file.rs [hunk 2]
```diff
@@ -293,7 +294,10 @@ impl ConfigFile {
   }
 
   pub fn new(text: &str, path: &Path) -> Result<Self, AnyError> {
-    let jsonc = jsonc_parser::parse_to_serde_value(text)?.unwrap();
+    let jsonc = match jsonc_parser::parse_to_serde_value(text) {
+      Ok(Some(value)) if value.is_object() => value,
+      _ => return Err(anyhow!("Unable to parse config file JSON")),
```
Comment (kitsonk) [inline cli/config_file.rs:299]: Would be better to provide the path as as part of the error message.

**Review by kitsonk — COMMENTED** [review]
Thanks for the PR, one thought/suggestion.

File: cli/config_file.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (romainPrignon) [reply cli/config_file.rs:299]: And only throw for unparsable files ?

Comment (romainPrignon) [reply cli/config_file.rs:299]: We might also be consistent with import_map error. I used the same error message.
If we plan to update this one, We may also update import_map

**Comment by romainPrignon** [comment]
Ok, now we emit error if config file is not parsable or not a json object
Empty or commented json file are OK
I updated the error message (see test)

Let me know what you think

File: cli/config_file.rs (hunk 2 above)
Comment (kitsonk) [reply cli/config_file.rs]: Yeah, we should improve that. But that would be a seperate PR/issue. Opened #10768.

**Comment by kitsonk** [comment]
Looks like we need to run `./tools/format.js` to format the changes.

**Comment by romainPrignon** [comment]
My bad, it should be ok now.
Sorry for the context switching and manually having to run the workflow

**Review by kitsonk — APPROVED** [review]
LGTM, thanks for the contribution.

**Pull request merged**